luda-editor/server/src/utils/mod.rs

```diff
@@ -9,3 +9,43 @@ macro_rules! simple_error_impl {
         impl std::error::Error for $error_struct {}
     };
 }
+
+#[macro_export]
+macro_rules! append_slash {
+    ($($x:expr),+ $(,)?) => {{
+        let mut result = String::new();
+        $(
+            let x = $x.to_string();
+            if result.is_empty() {
+                result = x;
+            } else if result.ends_with('/') {
+                if x.starts_with('/') {
+                    result.push_str(&x[1..]);
+                } else {
+                    result.push_str(&x);
+                }
+            } else {
+                if x.starts_with('/') {
+                    result.push_str(&x);
+                } else {
+                    result.push('/');
+                    result.push_str(&x);
+                }
+            }
+        )+
+        result
+    }};
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_append_slash() {
+        assert_eq!(append_slash!["a", "b"], "a/b");
+        assert_eq!(append_slash!["a/", "b"], "a/b");
+        assert_eq!(append_slash!["a", "/b"], "a/b");
+        assert_eq!(append_slash!["a/", "/b"], "a/b");
+        assert_eq!(append_slash!["a/", "/b/"], "a/b/");
+        assert_eq!(append_slash!["a/", "b/"], "a/b/");
+    }
+}
```

Comment:
It's duplicated with `luda-editor/client/src/storage/resource_url.rs`.
Do you think it is overhead to split macro into different crate?

Reply:
I'm not sure right now. It's the first and single case so I will make a decision when I make next second duplicated code.